src/common/hooks/useOnClickOutside/useOnClickOutside.ts

```diff
@@ -0,0 +1,18 @@
+import { RefObject, useEffect } from 'react';
+
+export const useOnClickOutside = (ref: RefObject<HTMLElement>, handler: (event: MouseEvent) => void) => {
+  useEffect(() => {
+    const listener = (event: MouseEvent) => {
+      if (!ref.current || ref.current.contains(event.target as Node)) {
+        return;
+      }
+      handler(event);
+    };
+
+    document.addEventListener('mousedown', listener);
+
+    return () => {
+      document.removeEventListener('mousedown', listener);
```

Comment:
```suggestion
      document.removeEventListener('mousedown', listener);
      document.removeEventListener('touchstart', listener);
```

The addition of document.addEventListener('touchstart', listener); alongside the existing mousedown event listener, means the hook now also listens for the beginning of touch interactions. This ensures that the hook works effectively on devices with touch interfaces, responding to any touch that starts outside the referenced element by invoking the handler function.